udata_datalake_service/background_tasks.py

```diff
@@ -0,0 +1,71 @@
+import logging
+import os
+import tempfile
+
+import boto3
+from celery import Celery
+import requests
+from botocore.client import Config
+from botocore.exceptions import ClientError
+from dotenv import load_dotenv
+
+from udata_datalake_service.producer import produce
+
+load_dotenv()
+
+BROKER_URL = os.environ.get("BROKER_URL", "redis://localhost:6380/0")
+MINIO_FOLDER = os.environ.get("MINIO_FOLDER", "folder")
+celery = Celery('tasks', broker=BROKER_URL)
+
+
+def download_resource(url):
+    tmp_file = tempfile.NamedTemporaryFile(delete=False)
+    with requests.get(url, stream=True) as r:
+        r.raise_for_status()
+        for chunk in r.iter_content(chunk_size=1024):
+            tmp_file.write(chunk)
+    tmp_file.close()
+    return tmp_file
+
+
+def save_resource_to_minio(resource_file, key, resource):
+    logging.info("Saving to minio")
+    s3 = boto3.client(
+        "s3",
+        endpoint_url=os.getenv("MINIO_URL"),
+        aws_access_key_id=os.getenv("MINIO_USER"),
+        aws_secret_access_key=os.getenv("MINIO_PWD"),
+        config=Config(signature_version="s3v4"),
+    )
+    try:
+        with open(resource_file.name, "rb") as f:
+            s3.upload_fileobj(f, os.getenv("MINIO_BUCKET"), MINIO_FOLDER + key + "/" + resource["id"])
```

Comment:
We save file without file extension in its name. I wonder how this would be painful in the future. It means that without external knowledge (as API info), user can't know how to open it. 
I think we could let this that way for now but we have to be aware that it could be a problem in future development.